ui/pages/confirmations/components/confirm/info/shared/redesigned-gas-fees/redesigned-gas-fees.tsx

```diff
@@ -0,0 +1,333 @@
+            conversionRate,
+            fromCurrency: EtherDenomination.GWEI,
+            toCurrency: currentCurrency,
+            numberOfDecimals: 2,
+          }),
+        ),
+      )
+    : null;
+
+  return (
+    <>
+      <ConfirmInfoRow
+        label="Estimated fee"
+        variant={ConfirmInfoRowVariant.Default}
+        tooltip="estimated fee tooltip"
+      >
+        <Box
+          display={Display.Flex}
+          flexDirection={FlexDirection.Row}
+          justifyContent={JustifyContent.spaceBetween}
+          alignItems={AlignItems.center}
+          textAlign={TextAlign.Center}
+          style={{ flexGrow: '1' }}
+        >
+          <Box
+            display={Display.Flex}
+            flexDirection={FlexDirection.Row}
+            justifyContent={JustifyContent.spaceEvenly}
+            alignItems={AlignItems.center}
+            textAlign={TextAlign.Center}
+            style={{ flexGrow: '1' }}
+          >
+            <Text color={TextColor.textAlternative}>{currentCurrencyFees}</Text>
+            <Text color={TextColor.textAlternative}>{nativeCurrencyFees}</Text>
+          </Box>
+
+          <EditGasIcon
+            supportsEIP1559={supportsEIP1559}
+            setShowCustomizeGasPopover={setShowCustomizeGasPopover}
+          />
+        </Box>
+      </ConfirmInfoRow>
+
+      {supportsEIP1559 && (
+        <ConfirmInfoRow
+          label="Gas speed"
+          variant={ConfirmInfoRowVariant.Default}
+        >
+          <Box display={Display.Flex} alignItems={AlignItems.center}>
+            {/* TODO: Fix bug in the gas timing component after selection is made */}
+            <GasTiming
+              maxFeePerGas={String(maxFeePerGas)}
+              maxPriorityFeePerGas={String(maxPriorityFeePerGas)}
+            />
+          </Box>
+        </ConfirmInfoRow>
+      )}
+
+      {/* TODO: Add separator */}
+
+      <ConfirmInfoRow
+        label="Total"
+        variant={ConfirmInfoRowVariant.Default}
+        tooltip="total tooltip"
+      >
+        <Text>{nativeCurrencyFees}</Text>
+      </ConfirmInfoRow>
+
+      {/* L2 Fees Breakdown */}
+
+      {hasLayer1GasFee && (
+        <Box
+          padding={4}
+          display={Display.Flex}
+          alignItems={AlignItems.center}
+          justifyContent={JustifyContent.center}
+        >
+          <Button
+            style={{ textDecoration: 'none' }}
+            size={ButtonSize.Sm}
+            variant={ButtonVariant.Link}
+            endIconName={
+              expandFeeDetails ? IconName.ArrowUp : IconName.ArrowDown
+            }
+            color={IconColor.primaryDefault}
+            data-testid="expand-fee-details-button"
+            onClick={() => setExpandFeeDetails(!expandFeeDetails)}
+          >
+            <Text variant={TextVariant.bodySm} color={IconColor.primaryDefault}>
+              {'feeDetails'}
+            </Text>
+          </Button>
+        </Box>
+      )}
+
+      {hasLayer1GasFee && expandFeeDetails && (
```

Comment:
Given the size of this function, should we break it up a little into local components such as `Layer1Fees`, `TotalRow`, `EstimatedFeeRow`, `GasTimings` etc.

Not just the JSX but the associated logic also so it can be encapsulated alongside the relevant rendering.